store/cachemulti/store.go

```diff
@@ -123,15 +123,22 @@ func (cms Store) GetStoreType() types.StoreType {
 // Write calls Write on each underlying store.
 func (cms Store) Write() {
 	cms.db.Write()
-	wg := sync.WaitGroup{}
-	wg.Add(len(cms.stores))
+	eg := new(errgroup.Group)
 	for _, store := range cms.stores {
-		go func(s types.CacheWrap) {
-			defer wg.Done()
+		s := store // https://golang.org/doc/faq#closures_and_goroutines
+		eg.Go(func() (err error) {
+			defer func() {
+				if r := recover(); r != nil {
+					err = fmt.Errorf("panic in Write: %v", r)
+				}
+			}()
 			s.Write()
-		}(store)
+			return nil
+		})
+	}
+	if err := eg.Wait(); err != nil {
```

Comment:
Not sure if the panic is handled somewhere else in case: this is not deterministic for multiple errors.

In the godoc for Wait: 
> returns the first non-nil error (if any) from them.